New sdk configuration (#245)
DataDog/dd-sdk-reactnative: docs/new_setup.md

<!-- Note: this file is temporary and should be integrated in the additional setup docs once that is merged -->

## Initializing asynchronously

> If your app includes a lot of animations at its start, running code during these animations might delay them on some devices. We provide a simple way to delay the Datadog React Native SDK for RUM aynchronously.

Reviewer: ```suggestion
If your app includes a lot of animations when it starts, running code during these animations might delay them on some devices. To delay the Datadog React Native SDK for RUM asynchronously, set the `initializationMode` to `InitializationMode.ASYNC` in your configuration:
```

> Modify your configuration to set the `initializationMode` to `InitializationMode.ASYNC`:

Reviewer: ```suggestion
```

> 

Reviewer: ```suggestion
```

```js
import {
    DatadogProvider,
    DatadogProviderConfiguration,
    InitializationMode
} from '@datadog/mobile-react-native';

const datadogConfiguration = new DatadogProviderConfiguration(
    '<CLIENT_TOKEN>',
    '<ENVIRONMENT_NAME>',
    '<RUM_APPLICATION_ID>',
    true,
    true,
    true
);
datadogConfiguration.initializationMode = InitializationMode.ASYNC;

export default function App() {
    return (
        <DatadogProvider configuration={datadogConfiguration}>
            <Navigation />
        </DatadogProvider>
    );
}
```

> All interactions with the RUM sdk (view tracking, actions, resources tracing, etc.) will still be recorded and kept in a queue to a limit of 50 events.

Reviewer: ```suggestion
All interactions with the RUM SDK (view tracking, actions, resources tracing, and so on) is still recorded and kept in a queue with a limit of 50 events.
```

> Logs won't be recorded and calling a `DdLogs` method before the actual initialization might break logging.

Reviewer: ```suggestion
Logs are not recorded and calling a `DdLogs` method before the actual initialization might break logging.
```

## Delaying the initialization

> You may want to wait before initializing the SDK, for instance if you wish to use a different configuration based on the user role, or to fetch the configuration from a server.

Reviewer: ```suggestion
There are situations where you want to wait before initializing the SDK, for instance, if you want to use a different configuration based on the user role or to fetch the configuration from a server.
```

> We provide a way to auto-instrument your app from the start (automatically collecting user interactions, XHR resources and errors) and record up to 50 RUM events before initializing the SDK.

Reviewer: ```suggestion
In that case, you can auto-instrument your app from the start (automatically collect user interactions, XHR resources, and errors) and record up to 50 RUM events before initializing the SDK.
```

```js
import {
    DatadogProvider,
    DatadogProviderConfiguration,
    InitializationMode
} from '@datadog/mobile-react-native';

const datadogAutoInstrumentation = {
    trackErrors: true,
    trackInteractions: true,
    trackResources: true,
    firstPartyHosts: [''],
    resourceTracingSamplingRate: 100
};

const initializeApp = async () => {

>     const configuration = await fetchDatadogConfiguration();

Reviewer 2: What's `fetchDatadogConfiguration` in this context?

    await DatadogProvider.initialize(configuration);
};

export default function App() {
    useEffect(() => initializeApp(), []);

    return (
        <DatadogProvider configuration={datadogAutoInstrumentation}>
            <Navigation />
        </DatadogProvider>
    );
}
```

> where your configuration has the following keys:

Reviewer: ```suggestion
Where your configuration has the following keys:
```

<!-- Note: this file is temporary and should be integrated in the additional setup docs once that is merged -->

## Initializing asynchronously

If your app includes a lot of animations when it starts, running code during these animations might delay them on some devices. To delay the Datadog React Native SDK for RUM asynchronously, set the `initializationMode` to `InitializationMode.ASYNC` in your configuration:

```js
import {
    DatadogProvider,
    DatadogProviderConfiguration,
    InitializationMode
} from '@datadog/mobile-react-native';

const datadogConfiguration = new DatadogProviderConfiguration(
    '<CLIENT_TOKEN>',
    '<ENVIRONMENT_NAME>',
    '<RUM_APPLICATION_ID>',
    true,
    true,
    true
);
datadogConfiguration.initializationMode = InitializationMode.ASYNC;

export default function App() {
    return (
        <DatadogProvider configuration={datadogConfiguration}>
            <Navigation />
        </DatadogProvider>
    );
}
```

> All interactions with the RUM SDK (view tracking, actions, resources tracing, and so on) is still recorded and kept in a queue with a limit of 50 events.

Reviewer: ```suggestion
All interactions with the RUM SDK (view tracking, actions, resources tracing, and so on) are still recorded and kept in a queue with a limit of 50 events.
```

> Logs are not recorded and calling a `DdLogs` method before the actual initialization might break logging.

Author: We need to enable custom timestamps in logs first. I can take care of that in a following release.

> Logs are not recorded and calling a `DdLogs` method before the actual initialization might break logging.

Reviewer 3: Are we saying as a consequence logs won't work during the application's lifetime or only before initialization?

> Logs are not recorded and calling a `DdLogs` method before the actual initialization might break logging.

Author: As a consequence logs won't work during the application's lifetime

## Delaying the initialization

> You may want to wait before initializing the SDK, for instance if you wish to use a different configuration based on the user role, or to fetch the configuration from one of your servers.

Reviewer: ```suggestion
There may be situations where you want to wait before initializing the SDK. For example, when you want to use a different configuration based on the user role or to fetch the configuration from one of your servers.
```

> In that case, you can auto-instrument your app from the start (automatically collect user interactions, XHR resources, and errors) and record up to 50 RUM events before initializing the SDK.

Author: ```suggestion
In that case, you can auto-instrument your app from the start (automatically collect user interactions, XHR resources, and errors) and record up to 100 RUM events before initializing the SDK.
```
